anonlink/similarities/_dice_python.py

```diff
@@ -64,5 +65,21 @@ def dice_coefficient_python(
         result_sims.extend(sim for _, sim in top_k)
         result_indices0.extend(repeat(i, len(top_k)))
         result_indices1.extend(j for j, _ in top_k)
+
+    np_sims = np.frombuffer(result_sims, dtype=result_sims.typecode)
```

Comment:
If we have to use numpy - why not use numpy arrays from the start?

Reply:
NumPy arrays are not dynamic. This makes it harder to use them to store a variable number of candidates.

The problem with just casting a Python array to NumPy is that NumPy won't take ownership of the memory, so we'd have issues with garbage collection.